arbiter-core/src/environment.rs

```diff
@@ -295,7 +295,31 @@ impl Environment {
                         let mut guard = lock.lock().map_err(|e| EnvironmentError::Pause {
                             cause: format!("{:?}", e),
                         })?;
-                        while state.load(std::sync::atomic::Ordering::Relaxed) == State::Paused {
+
+                        // this logic here ensures we catch any edge case last transactions and send
+                        // the appropriate error so that we dont hang in
+                        // limbo forever
+                        while let Ok((_, _, sender)) = tx_receiver.try_recv() {
+                            let error_outcome =
+                                TransactionOutcome::Error(EnvironmentError::Pause {
+                                    cause: "Environment is paused".into(),
+                                });
+                            let revm_result = RevmResult {
+                                outcome: error_outcome,
+                                block_number: convert_uint_to_u64(evm.env.block.number).map_err(
+                                    |e| EnvironmentError::Conversion {
+                                        cause: format!("{:?}", e),
+                                    },
+                                )?,
+                            };
+                            sender.send(revm_result).map_err(|e| {
+                                EnvironmentError::Communication {
+                                    cause: format!("{:?}", e),
+                                }
+                            })?;
+                        }
```

Comment:
This should work I think!

Reply:
Just double checking the logic:
1. We set the state to paused in some way via the `AtomicState`.
2. This eventually hits the `send_transaction()` and `call()` function conditional checks and will eventually error out the transactions/calls that users are trying to send.
3. All the while we also eventually end up in this `Paused` state and enter the `while let` on the `try_recv()` and consume all the possible txs in the queue.
4. Now we rest easy until the condvar is signaled to restart the environment.

Sound right?

Reply:
Thats exactly right!